### PR TITLE
Add reducer/action unit tests

### LINKED ISSUE #148: Visualizer - Add Support For Code Coverage Checking
**Is your feature request or enhancement related to a problem? What specifically could be better? Please describe.**
We should add support for code coverage checking to the typescript code that lives in the visualizer. 

**Describe the solution you'd like**
One each build, Travis CI should automatically upload code coverage information to codecov. Here is an example of how to do this for typescript: https://github.com/codecov/example-typescript

### LINKED ISSUE #215: Add testing for redux store
**Is your feature request or enhancement related to a problem? What specifically could be better? Please describe.**
We want to add unit tests for all actions, reducers, sagas and selectors in redux

### PR DESCRIPTION
**Pull request by kelvinkoon**
### Description
Unit testing has been added for the Visualizer's actions and reducers. Bash scripts have also been added to start the test scripts from any directory.

### Testing Done
Running `yarn test` shows the 21 tests from Jest passing. 

### Resolved Issues
Resolves #215  (except for sagas, which will be added later)

### Review Checklist
***It is the reviewers responsibility to also make sure every item here has been covered***
- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom` 
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

***Feel free to make additions of things that we should be checking to this file if you think there's something missing!!!!***

### CODE REVIEW

File: src/corner_kick/src/store/reducers/__tests__/ros.test.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            expect(state.params).toEqual(['dongle_params', 'camera_params']);
+        });
+    });
+    describe('when we receive other actions', () => {
+        it('should return state from action payload', () => {
+            const mockAction = ros.newMessage('topic', 'message');
+
+            const state = rosReducer(undefined, mockAction);
+
+            expect(state).toEqual(state);
```
Comment (qgolsteyn) [inline src/corner_kick/src/store/reducers/__tests__/ros.test.ts]: Doesn't do anything. You are comparing `state` with itself. Maybe create a state object, pass it to the reducer and verify if the return value is the same.

**Review by qgolsteyn — CHANGES_REQUESTED** [review]
Looks good! Just a small issue near the end, but other than that good work!

File: src/corner_kick/src/store/reducers/__tests__/ros.test.ts (hunk 1 above)
Comment (kelvinkoon) [reply src/corner_kick/src/store/reducers/__tests__/ros.test.ts]: Updated to pass in a mockState object and compare it with the returned state.

File: src/corner_kick/package.json [hunk 2]
```diff
@@ -30,6 +31,7 @@
         "html-webpack-plugin": "3.2.0",
         "http-server": "0.11.1",
         "jest": "23.6.0",
+        "redux-mock-store": "^1.5.3",
```
Comment (qgolsteyn) [inline src/corner_kick/package.json]:
``` suggestion
        "redux-mock-store": "1.5.3",
```

**Review by qgolsteyn — CHANGES_REQUESTED** [review]
Missed a small thing on my first review sorry :frowning_face: otherwise, looks good!

File: src/corner_kick/package.json (hunk 2 above)
Comment (qgolsteyn) [reply src/corner_kick/package.json]: You should be able to accept the suggestion from GitHub

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:

**Review by qgolsteyn — APPROVED** [review]
LGTM :+1:

**Pull request merged**